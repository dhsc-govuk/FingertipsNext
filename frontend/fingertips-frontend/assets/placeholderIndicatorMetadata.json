{
    "indicatorID": "placeholder indicator id",
    "indicatorName": "placeholder indicator name",
    "indicatorDefinition": "placeholder indicator definition",
    "dataSource": "placeholder data source",
    "earliestDataPeriod": 1234,
    "latestDataPeriod": 5678,
    "lastUpdatedDate": "1066-10-14T12:00:00",
    "unitLabel": "placeholder unit label",
    "valueType": "placeholder value type",
    "denominatorType": "placeholder denominator type",
    "ciMethod": "placeholder ci method",
    "unitValue": "placeholder unit value",
    "rationale": "placeholder rationale",
    "method": "placeholder method",
    "standardPopulation": "placeholder standard population",
    "ciMethodDetails": "placeholder ci method details",
    "countSource": "placeholder numerator source",
    "countDefinition": "placeholder numerator definition\n Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque venenatis sapien eget nisl dapibus sodales",
    "denominatorSource": "placeholder denominator source",
    "denominatorDefinition": "placeholder denominator definition\n Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque venenatis sapien eget nisl dapibus sodales",
    "disclouseControl": "placeholder disclouse (disclosure) control",
    "caveats": "placeholder caveats",
    "copyright": "placeholder copyright",
    "reuse": "placeholder reuse",
    "notes": "placeholder notes",
    "frequency": "placeholder frequency",
    "rounding": "placeholder rounding",
    "polarity": "placeholder polarity",
    "useProportionsForTrend": true,
    "benchmarkComparisonMethod": "placeholder benchmark comparison method",
    "usedInPoc": false
}