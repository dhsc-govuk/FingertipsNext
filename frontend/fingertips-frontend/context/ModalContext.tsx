import {
  createContext,
  FC,
  ReactNode,
  useContext,
  useMemo,
  useState,
} from 'react';
import { ModalPane } from '@/components/molecules/ModalPane/ModalPane';

export interface Modal {
  content?: ReactNode;
  title?: string;
}

export const ModalContext = createContext<{
  modal: Modal;
  setModal: (newPreview: Modal) => void;
}>({ modal: {}, setModal: () => undefined });

export const useModal = () => {
  const { modal, setModal } = useContext(ModalContext);
  const { content, title } = modal;
  return { modal, setModal, content, title };
};

interface ModalProviderProps {
  children: ReactNode;
}

export const ModalProvider: FC<ModalProviderProps> = ({ children }) => {
  const [modal, setModal] = useState<Modal>({});
  const providerValue = useMemo(
    () => ({
      modal,
      setModal,
    }),
    [modal]
  );

  return (
    <ModalContext.Provider value={providerValue}>
      {children}
      <ModalPane />
    </ModalContext.Provider>
  );
};
