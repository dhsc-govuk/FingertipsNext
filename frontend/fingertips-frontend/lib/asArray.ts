export const asArray = (value?: string | string[]): string[] => {
  if (value) {
    return Array.isArray(value) ? value : [value];
  }
  return [];
};
