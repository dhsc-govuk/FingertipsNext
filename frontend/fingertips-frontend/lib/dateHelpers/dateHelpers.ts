export function isWithinOneMonth(
  currentDate: Date,
  indicatorDate: Date
): boolean {
  const oneMonthAgo = new Date(currentDate);
  oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);

  return oneMonthAgo <= indicatorDate;
}

export function formatDate(date: Date | undefined): string {
  if (!date) return 'unknown';
  const day = String(date.getDate()).padStart(2, '0');
  const month = date.toLocaleString('en-GB', { month: 'long' });
  const year = date.getFullYear();

  return `${day} ${month} ${year}`;
}
