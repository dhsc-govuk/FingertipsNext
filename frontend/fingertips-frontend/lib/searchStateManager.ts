import { asArray } from './pageHelpers';

export enum SearchParams {
  SearchedIndicator = 'si',
  IndicatorsSelected = 'is',
  AreasSelected = 'as',
  AreaTypeSelected = 'ats',
  GroupTypeSelected = 'gts',
  ConfidenceIntervalSelected = 'cis',
}

export type SearchParamKeys = `${SearchParams}`;

const multiValueParams = [
  SearchParams.IndicatorsSelected as string,
  SearchParams.AreasSelected as string,
  SearchParams.ConfidenceIntervalSelected as string,
];

export type SearchStateParams = {
  [SearchParams.SearchedIndicator]?: string;
  [SearchParams.IndicatorsSelected]?: string[];
  [SearchParams.AreasSelected]?: string[];
  [SearchParams.AreaTypeSelected]?: string;
  [SearchParams.GroupTypeSelected]?: string;
  [SearchParams.ConfidenceIntervalSelected]?: string[];
};

const isMultiValueTypeParam = (searchParamKey: SearchParamKeys) =>
  multiValueParams.includes(searchParamKey);

export class SearchStateManager {
  private searchState: SearchStateParams;
  private searchStateParams: URLSearchParams;

  constructor(searchState: SearchStateParams = {}) {
    this.searchState = {};
    this.searchStateParams = new URLSearchParams();

    this.searchState = searchState;
  }

  private constructPath(path: string) {
    if (this.searchStateParams.size === 0) {
      return path;
    }
    return `${path}?${this.searchStateParams.toString()}`;
  }

  private addParamToPath(searchParamKey: SearchParamKeys) {
    const paramsToAdd = asArray(this.searchState[searchParamKey]);

    if (paramsToAdd.length > 0) {
      paramsToAdd.forEach((paramToAdd) => {
        this.searchStateParams.append(searchParamKey, paramToAdd);
      });
    }
  }

  public addParamValueToState(
    searchParamKey: SearchParamKeys,
    paramToAdd: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = this.searchState[searchParamKey];

      const currentParamStateAsArray = asArray(currentParamState);
      const doesParamExist = currentParamStateAsArray.find(
        (currentParam) => currentParam === paramToAdd
      );

      if (!doesParamExist) {
        const newState: SearchStateParams = {
          ...this.searchState,
          [searchParamKey]: [...currentParamStateAsArray, paramToAdd],
        };

        this.searchState = newState;
      }
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: paramToAdd,
      };

      this.searchState = newState;
    }
  }

  public removeParamValueFromState(
    searchParamKey: SearchParamKeys,
    paramToRemove?: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = asArray(this.searchState[searchParamKey]);

      const newParamState = currentParamState?.filter((paramValue) => {
        return paramValue !== paramToRemove;
      });

      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: newParamState,
      };

      this.searchState = newState;
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: undefined,
      };

      this.searchState = newState;
    }
  }

  public removeAllParamFromState(searchParamKey: SearchParamKeys) {
    this.searchState[searchParamKey] = undefined;
  }

  public static setStateFromParams(params: URLSearchParams) {
    const newState = Object.values(SearchParams).reduce<SearchStateParams>(
      (state, searchParamKey) => {
        if (isMultiValueTypeParam(searchParamKey)) {
          const paramValues = params.getAll(searchParamKey);

          if (paramValues.length > 0) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValues,
            };

            return newState;
          }
        } else {
          const paramValue = params.get(searchParamKey);

          if (paramValue) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValue,
            };

            return newState;
          }
        }
        return state;
      },
      {}
    );

    const searchStateManager = new SearchStateManager(newState);
    return searchStateManager;
  }

  public static setStateFromSearchStateParams(params?: SearchStateParams) {
    if (params) {
      const newState = Object.values(SearchParams).reduce<SearchStateParams>(
        (state, searchParamKey) => {
          if (isMultiValueTypeParam(searchParamKey)) {
            const paramValues = asArray(params[searchParamKey]);

            if (paramValues.length > 0) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValues,
              };

              return newState;
            }
          } else {
            const paramValue = params[searchParamKey];

            if (paramValue) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValue,
              };

              return newState;
            }
          }
          return state;
        },
        {}
      );

      const searchStateManager = new SearchStateManager(newState);
      return searchStateManager;
    }
    return new SearchStateManager();
  }

  public getSearchState() {
    return this.searchState;
  }

  public generatePath(path: string) {
    this.searchStateParams = new URLSearchParams();

    Object.values(SearchParams).forEach((searchParamKey) => {
      this.addParamToPath(searchParamKey);
    });

    return this.constructPath(path);
  }
}
