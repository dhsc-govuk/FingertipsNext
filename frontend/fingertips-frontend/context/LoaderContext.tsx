'use client';

import React, {
  ReactNode,
  createContext,
  useContext,
  useMemo,
  useState,
  useEffect,
  Suspense,
  Dispatch,
  SetStateAction,
} from 'react';
import { usePathname } from 'next/navigation';
import { LoadingBox } from 'govuk-react';
import { useSearchStateParams } from '@/components/hooks/useSearchStateParams';
import { SearchStateParams } from '@/lib/searchStateManager';

export type LoaderContext = {
  getIsLoading: () => boolean;
  setIsLoading: (loading: boolean) => void;
};

type LoaderContextProvider = {
  children: ReactNode;
};

const LoaderContext = createContext<LoaderContext | undefined>(undefined);

export const LoaderProvider: React.FC<LoaderContextProvider> = ({
  children,
}) => {
  const pathname = usePathname();
  const [searchState, setSearchState] = useState<SearchStateParams>({});

  const [isLoading, setIsLoading] = useState<boolean>(true);

  const contextValue: LoaderContext = useMemo(
    () => ({
      getIsLoading: () => isLoading,
      setIsLoading,
    }),
    [isLoading, setIsLoading]
  );

  useEffect(() => {
    setIsLoading(false);
  }, [setIsLoading, pathname, searchState]);

  return (
    <LoaderContext.Provider value={contextValue}>
      <Suspense>
        <SuspendedUseSearchStateParams setSearchState={setSearchState} />
      </Suspense>
      <LoadingBox loading={isLoading} timeIn={200} timeOut={200}>
        {children}
      </LoadingBox>
    </LoaderContext.Provider>
  );
};

export const useLoadingState = () => {
  const context = useContext(LoaderContext);

  if (!context) {
    throw new Error('useLoadingState must be used within a LoaderProvider');
  }

  return context;
};

// Next has a very peculiar quirk where useSearchState cannot be called server
// side, which works fine during dev but build will fail.
// Here we're creating a component which will not render server side by
// suspending it until client mounts at which point it can set the loading
// context value, all this is more complicated because loader context is in a
// layout.
function SuspendedUseSearchStateParams({
  setSearchState,
}: Readonly<{ setSearchState: Dispatch<SetStateAction<SearchStateParams>> }>) {
  const searchParams = useSearchStateParams();
  useEffect(() => {
    setSearchState(searchParams);
  }, [searchParams, setSearchState]);

  return null;
}
