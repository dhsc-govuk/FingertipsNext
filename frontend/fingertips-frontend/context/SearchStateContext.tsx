'use client';

import React, { ReactNode, createContext, useContext, useState } from 'react';
import { SearchStateParams } from '@/lib/searchStateManager';

export type SearchStateContext = {
  getSearchState: () => SearchStateParams;
  setSearchState: (state: SearchStateParams) => void;
};

type SearchStateContextProvider = {
  children: ReactNode;
};

const SearchStateContext = createContext<SearchStateContext | undefined>(
  undefined
);

export const SearchStateProvider: React.FC<SearchStateContextProvider> = ({
  children,
}) => {
  const [searchState, setSearchState] = useState<SearchStateParams>({});

  const contextValue: SearchStateContext = {
    getSearchState: () => searchState,
    setSearchState,
  };

  return (
    <SearchStateContext.Provider value={contextValue}>
      {children}
    </SearchStateContext.Provider>
  );
};

export const useSearchState = () => {
  const context = useContext(SearchStateContext);

  if (!context) {
    throw new Error('useLoadingState must be used within a LoaderProvider');
  }

  return context;
};
