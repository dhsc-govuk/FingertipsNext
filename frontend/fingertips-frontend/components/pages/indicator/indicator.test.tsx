import { render, screen } from '@testing-library/react';
import { IndicatorDefinition, IndicatorDefinitionProps } from '.';
import placeholderIndicatorMetadata from '../../../assets/placeholderIndicatorMetadata.json';
import { formatDate } from '@/lib/dateHelpers/dateHelpers';
import { SearchParams } from '@/lib/searchStateManager';
import { LoaderContext } from '@/context/LoaderContext';
import { SearchStateContext } from '@/context/SearchStateContext';

const indicatorDefinition: IndicatorDefinitionProps = {
  ...placeholderIndicatorMetadata,
  indicatorID: String(placeholderIndicatorMetadata.indicatorID),
  lastUpdatedDate: new Date(placeholderIndicatorMetadata.lastUpdatedDate),
};

const searchState = {
  [SearchParams.IndicatorsSelected]: ['1', '2'],
  [SearchParams.AreasSelected]: ['A001'],
};

jest.mock('next/navigation', () => {
  const originalModule = jest.requireActual('next/navigation');

  return {
    ...originalModule,
    usePathname: jest.fn(),
  };
});

const mockSetIsLoading = jest.fn();
const mockLoaderContext: LoaderContext = {
  getIsLoading: jest.fn(),
  setIsLoading: mockSetIsLoading,
};
jest.mock('@/context/LoaderContext', () => {
  return {
    useLoadingState: () => mockLoaderContext,
  };
});

const mockSearchStateContext: SearchStateContext = {
  getSearchState: jest.fn(),
  setSearchState: jest.fn(),
};
jest.mock('@/context/SearchStateContext', () => {
  return {
    useSearchState: () => mockSearchStateContext,
  };
});

describe('contents items should link to appropriate headings', () => {
  beforeEach(() => {
    render(
      <IndicatorDefinition
        indicatorDefinitionProps={indicatorDefinition}
        searchState={searchState}
      />
    );
  });

  interface TestData {
    linkText: string;
    titleText: string;
  }

  it.each<TestData>([
    {
      linkText: 'Indicator definitions',
      titleText: 'Indicator definitions',
    },
    {
      linkText: 'Data sources and reuse',
      titleText: 'Data sources and reuse',
    },
    {
      linkText: 'Benchmarking and confidence information',
      titleText: 'Benchmarking and confidence information',
    },
    {
      linkText: 'Indicator rationale',
      titleText: 'Indicator rationale',
    },
    {
      linkText: 'Notes and caveats',
      titleText: 'Notes and caveats',
    },
    {
      linkText: 'Profiles and links',
      titleText: 'Public health profile usage and related content',
    },
  ])('$linkText', ({ linkText, titleText }) => {
    const href = screen
      .getByRole('link', { name: linkText })
      .getAttribute('href')!;

    expect(
      screen.getByRole('heading', { name: titleText }).closest(`${href}`)
    ).toHaveAttribute('id', href.slice(1));
  });
});

describe('indicator description page', () => {
  beforeEach(() => {
    render(
      <IndicatorDefinition
        indicatorDefinitionProps={indicatorDefinition}
        searchState={searchState}
      />
    );
  });

  it('should match snapshot', () => {
    const page = render(
      <IndicatorDefinition
        indicatorDefinitionProps={indicatorDefinition}
        searchState={searchState}
      />
    );

    expect(page.asFragment()).toMatchSnapshot();
  });

  it('should the back link path to the chart page with the searchState paths', () => {
    const expectedPath = [
      `/chart`,
      `?${SearchParams.IndicatorsSelected}=1&${SearchParams.IndicatorsSelected}=2`,
      `&${SearchParams.AreasSelected}=A001`,
    ].join('');

    const backLink = screen.getByRole('link', { name: /back/i });

    expect(backLink).toBeInTheDocument();

    expect(backLink.getAttribute('href')).toBe(expectedPath);
  });

  it('should lead with a title containing the indicator name', () => {
    expect(screen.getAllByRole('heading')[0]).toHaveTextContent(
      indicatorDefinition.indicatorName
    );
  });

  it('should contain the indicator ID', () => {
    expect(
      screen.getByText(indicatorDefinition.indicatorID, { exact: false })
    ).toBeInTheDocument();
  });

  it('should contain the indicator definition', () => {
    expect(
      screen.getByText(indicatorDefinition.indicatorDefinition, {
        exact: false,
      })
    ).toBeInTheDocument();
  });

  it('should contain the data source', () => {
    expect(
      screen.getByText(indicatorDefinition.dataSource, { exact: false })
    ).toBeInTheDocument();
  });

  it('should contain the last updated date', () => {
    expect(
      screen.getByText(formatDate(indicatorDefinition.lastUpdatedDate), {
        exact: false,
      })
    ).toBeInTheDocument();
  });
});
