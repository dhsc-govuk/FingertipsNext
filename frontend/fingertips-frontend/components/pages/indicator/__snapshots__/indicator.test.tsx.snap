// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`indicator description page > should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="src__StyledBackLink-sc-159gh4v-0 lgZUAq"
    data-testid="indicator-info-back-link"
    href="/chart?is=1&is=2&as=A001"
  >
    Back
  </a>
  <span
    class="src__Caption-sc-dd9g0r-0 hQkhRe"
  >
    Background information and indicator definitions for
  </span>
  <h1
    class="heading__StyledHeading-sc-1dsobun-0 eaMQId sc-gEvEer iypzoq"
    data-testid="indicator-name"
  >
    placeholder indicator name
  </h1>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 cALenh sc-aXZVg fyYrlG"
  >
    <p>
      Indicator ID placeholder indicator id
    </p>
  </div>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 cALenh sc-aXZVg fyYrlG"
  >
    <p>
      Last updated 14 October 1066
    </p>
  </div>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 cALenh sc-aXZVg fyYrlG"
  >
    <p>
      Frequency placeholder frequency
    </p>
  </div>
  <hr
    class="src__SectionBreak-sc-fayw97-0 sc-eqUAAy kMqhKb fBGttu"
  />
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb sc-aXZVg fyYrlG"
  >
    <p>
      Contents
    </p>
  </div>
  <ul
    class="src__OrderedList-sc-1392v8j-0 gWgAlH"
  >
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#definitions"
      >
        Indicator definitions
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#sources"
      >
        Data sources and reuse
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#benchmarking"
      >
        Benchmarking and confidence information
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#rationale"
      >
        Indicator rationale
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#notes"
      >
        Notes and caveats
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#links"
      >
        Profiles and links
      </a>
    </li>
  </ul>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="definitions"
  >
    Indicator definitions
  </h3>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Definition
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder indicator definition. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean dictum tempus molestie. Proin orci felis, ullamcorper auctor dui a, dictum luctus nisi. Integer dolor orci, pretium a lacinia quis, elementum a dui. Nunc a finibus massa, non imperdiet ex. Nunc pulvinar, lacus at egestas aliquam, ex mi semper odio, id molestie nulla metus sed leo.
    </p>
  </div>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Methodology
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder method
    </p>
  </div>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Definition of numerator
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder numerator definition
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque venenatis sapien eget nisl dapibus sodales
    </p>
  </div>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Definition of denominator
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder denominator definition
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque venenatis sapien eget nisl dapibus sodales
    </p>
  </div>
  <hr
    class="src__SectionBreak-sc-fayw97-0 sc-eqUAAy jCLFCp fBGttu"
  />
  <table
    class="src__StyledTable-sc-16s660v-0 duOufa"
  >
    <tbody
      class="TableBody-sc-1qqarm8-0"
    >
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Age
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder age information
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Sex
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder sex information
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Value Type
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder value type
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Unit
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder unit label
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Year type
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder year type (e.g. financial)
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Standard population/values
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder standard population
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Disclosure control
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder disclosure control
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Rounding
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder rounding
        </td>
      </tr>
    </tbody>
  </table>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="sources"
  >
    Data sources and reuse
  </h3>
  <table
    class="src__StyledTable-sc-16s660v-0 duOufa"
  >
    <tbody
      class="TableBody-sc-1qqarm8-0"
    >
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Data source
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder data source
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Current indicator source
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          <a
            class="src__Link-sc-1loawqx-0 jjZtWC"
            href="#"
          >
            placeholder current indicator source
          </a>
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Previous indicator source
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          <a
            class="src__Link-sc-1loawqx-0 jjZtWC"
            href="#"
          >
            placeholder previous indicator source
          </a>
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Source of numerator
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder numerator source
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Source of denominator
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder denominator source
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Copyright
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder copyright
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Date reuse
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder reuse
        </td>
      </tr>
    </tbody>
  </table>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="benchmarking"
  >
    Benchmarking and confidence information
  </h3>
  <table
    class="src__StyledTable-sc-16s660v-0 duOufa"
  >
    <tbody
      class="TableBody-sc-1qqarm8-0"
    >
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Benchmarking method
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder benchmarking method
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Benchmarking significance level
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder benchmarking signficance level
        </td>
      </tr>
      <tr
        class="Row-sc-1y8eqfp-0"
      >
        <th
          class="Cell-sc-msv0l6-0 iftjBc"
        >
          Confidence interval method
        </th>
        <td
          class="Cell-sc-msv0l6-0 gaSlHc"
        >
          placeholder ci method
        </td>
      </tr>
    </tbody>
  </table>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Confidence interval methodology
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder ci method details. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean dictum tempus molestie. Proin orci felis, ullamcorper auctor dui a, dictum luctus nisi. Integer dolor orci, pretium a lacinia quis, elementum a dui. Nunc a finibus massa, non imperdiet ex. Nunc pulvinar, lacus at egestas aliquam, ex mi semper odio, id molestie nulla metus sed leo. Maecenas eu nunc tincidunt, placerat ante ac, venenatis lorem. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
Fusce bibendum vestibulum lacus, non bibendum tellus ullamcorper ut. Curabitur at ipsum leo. Proin convallis pulvinar mi vitae aliquet.
    </p>
  </div>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="rationale"
  >
    Indicator rationale
  </h3>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      Donec arcu purus, auctor quis dolor suscipit, malesuada cursus nulla. Mauris et massa et libero tempus fringilla. Donec consectetur quis lorem sit amet lacinia. Maecenas non urna a risus gravida rutrum in sed lacus. In pharetra ultricies nisl vitae ultricies. Nulla ullamcorper et lectus eget imperdiet. Phasellus vel rutrum ex.
Suspendisse ligula orci, vestibulum non tincidunt at, congue in quam. Sed imperdiet erat vitae metus tincidunt imperdiet. Donec eget nisi vitae purus consequat gravida eu vel mauris. Nunc eros orci, facilisis in eleifend a, iaculis ut leo. Ut at venenatis ex, in aliquet leo. Proin ut bibendum leo, eu viverra magna.
    </p>
  </div>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="notes"
  >
    Notes and caveats
  </h3>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Notes
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder notes Aliquam accumsan dolor ut velit feugiat pellentesque. Mauris sagittis, orci sit amet sollicitudin finibus, velit tellus rhoncus purus, non convallis felis mi non ex. Fusce elementum lacus ac turpis faucibus eleifend. Integer elementum lorem vel leo luctus suscipit vel id tortor. Nam pulvinar lacus sit amet erat venenatis, non semper nunc sollicitudin.
    </p>
  </div>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Caveats
  </h4>
  <div
    class="src__StyledParagraph-sc-sgc9po-0 qWnWb"
  >
    <p>
      placeholder caveats  Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Pellentesque consequat magna sapien, ac dapibus arcu commodo vel. Nulla varius ut eros eget consectetur. Interdum et malesuada fames ac ante ipsum primis in faucibus. Phasellus posuere mollis dignissim.
    </p>
  </div>
  <h3
    class="heading__StyledHeading-sc-1dsobun-0 bRbIzR"
    id="links"
  >
    Public health profile usage and related content
  </h3>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    This indicator is used in the following health profiles
  </h4>
  <ul
    class="src__OrderedList-sc-1392v8j-0 kTfnIb"
  >
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder profile 1
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder profile 2
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder profile 3
      </a>
    </li>
  </ul>
  <h4
    class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
  >
    Related Content
  </h4>
  <ul
    class="src__OrderedList-sc-1392v8j-0 kTfnIb"
  >
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder link 1
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder link 2
      </a>
    </li>
    <li
      class="src__ListItem-sc-u4zf5o-0 fxpeqO"
    >
      <a
        class="src__Link-sc-1loawqx-0 jjZtWC"
        href="#"
      >
        placeholder link 3
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
